src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCharClass.cs

```diff
@@ -1356,158 +1308,13 @@ public static List<(char First, char Last)>? ComputeRanges(ReadOnlySpan<char> se
             return ranges;
         }
 
-        #region Perf workaround until https://github.com/dotnet/runtime/issues/61048 and https://github.com/dotnet/runtime/issues/59492 are addressed
-        // TODO: https://github.com/dotnet/runtime/issues/61048
-        // The below functionality needs to be removed/replaced/generalized.  The goal is to avoid relying on
-        // ToLower and culture-based operation at match time, and instead be able to compute at construction
-        // time case folding equivalence classes that let us determine up-front the set of characters considered
-        // valid for a match.  For now, we do this just for ASCII, and for anything else fall back to the
-        // pre-existing mechanism whereby a culture is used at construction time to ToLower and then one is
-        // used at match time to ToLower.  We also skip 'i' and 'I', as the casing of those varies across culture
-        // whereas every other ASCII value's casing is stable across culture.  We could hardcode the values for
-        // when an invariant vs tr/az culture vs any other culture is used, and we likely will, but for now doing
-        // so would be a breaking change, as in doing so we'd be relying only on the culture present at the time
-        // of construction rather than the one at the time of match.  That will be resolved with
-        // https://github.com/dotnet/runtime/issues/59492.
-
-        /// <summary>Creates a set string for a single character, optionally factoring in case-insensitivity.</summary>
+        /// <summary>Creates a set string for a single character.</summary>
         /// <param name="c">The character for which to create the set.</param>
-        /// <param name="caseInsensitive">null if case-sensitive; non-null if case-insensitive, in which case it's the culture to use.</param>
-        /// <param name="resultIsCaseInsensitive">false if the caller should strip out RegexOptions.IgnoreCase because it's now fully represented by the set; otherwise, true.</param>
         /// <returns>The create set string.</returns>
-        public static string OneToStringClass(char c, CultureInfo? caseInsensitive, out bool resultIsCaseInsensitive)
-        {
-            var vsb = new ValueStringBuilder(stackalloc char[4]);
+        public static string OneToStringClass(char c)
+            => CharsToStringClass(stackalloc char[1] { c });
```

Comment:
Doesn't have to be in this change, but subsequently we'll probably want to special-case this as well to return const class strings for ASCII.